### PR TITLE
support `getStaticProps` from theme config

### PR DESCRIPTION
**Pull request by B2o5T**
`getStaticProps` from the MDX page take priority over `getStaticProps` from the theme config

fixes https://github.com/shuding/nextra/issues/1227

### CODE REVIEW

**Comment by shuding** [comment]
Gonna improve this part a bit:

```js
  // `getStaticProps` from the MDX page take priority over `getStaticProps` from the theme config
  result.includes('getStaticProps')
```

**Comment by shuding** [comment]
I thought again about this, two major concerns I have:
- What should we do if the page itself has `getStaticProps`/`getServerSideProps`/`getStaticPaths`?
- Since we attach the same gSSP to all pages, we can't know which page is being rendered inside gSSP.

I have another similar proposal that can probably avoid those.

**Comment by B2o5T** [comment]
> Gonna improve this part a bit:
> 
> ```js
>   // `getStaticProps` from the MDX page take priority over `getStaticProps` from the theme config
>   result.includes('getStaticProps')
> ```

Since getStaticProps can be expected differently 

`export function getStaticProps`
`export const getStaticProps`
`export { getStaticProps } from './foo'`

The better way will be to convert the `result` into AST and parse it to find getStaticProps, but I guess it’s overkill

**Comment by shuding** [comment]
I pushed an commit to implement a `transform` method which can be very extendable

File: packages/nextra/src/loader.ts [hunk 1]
```diff
@@ -134,6 +135,15 @@ async function loader(
   // so we can generate the correct page map.
   context.addContextDependency(PAGES_DIR)
 
+  // Add local theme as a dependency
+  if (theme.startsWith('.') || theme.startsWith('/')) {
+    context.addDependency(path.resolve(theme))
+  }
+  // Add theme config as a dependency
+  if (themeConfig) {
+    context.addDependency(path.resolve(themeConfig))
```
Comment (B2o5T) [inline packages/nextra/src/loader.ts:144]: This is a nice improvement

**Pull request merged**